### ISSUE TITLE
tokio sleep need to run under tokio runtime

### ISSUE THREAD

**Issue by reyamir** [yukibtc/async-utility#1]
I'm facing a situation where my app crash because `tokio::sleep` is not running under `tokio runtime`. So I've made a fork and then change the `sleep fn` to this:
```rust
pub async fn sleep(duration: Duration) {
    #[cfg(not(target_arch = "wasm32"))]
    runtime().spawn_blocking(move || async move {
        tokio::time::sleep(duration).await;
    });
    #[cfg(target_arch = "wasm32")]
    gloo_timers::future::sleep(duration).await;
}
```

I wonder is this fine?

**Comment by reyamir** [comment]
I think will try this https://docs.rs/async-compat/latest/async_compat/ instead

**Comment by yukibtc** [comment]
Fixed the issue and published `async-utility` 0.3.1
